### PR TITLE
Feature/279 when built from Xcode, display something

### PR DESCRIPTION
**Pull request by terhechte**
#279 

This PR includes two indicators to show the user that Xi-Mac is being run from within Xcode:

First, a `[Debug] ..` string in the window titlebar:

screenshot 2018-10-03 at 18 35 05

Second, a different icon that kinda mimics the look of the Xcode icon:

xi-debug-512 copy

### CODE REVIEW

File: XiEditor/XiWindowController.swift [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import AppKit.NSWindowController
```
Comment (cmyr) [inline XiEditor/XiWindowController.swift]: You can copy over the header from one of the existing files (with the current year)

**Comment by terhechte** [comment]
@cmyr I broke something when I commited. The icon should also work now. I also added the license header.

**Comment by cmyr** [comment]
@terhechte awesome, this is working for me now.

My only lingering concern is that I'm seeing two new compiler errors (Xcode 10). The first is about your `#if debug` causing an unreachable line, and the other is about a downcast to `NSWindowController`. If you'd like to clean those up I'd be happy to merge.

Also it's a little embarrassing that our debug logo maybe looks a bit better than our default logo.. 😅💦 (aka, good job on the logo)

**Comment by terhechte** [comment]
@cmyr Thanks! Glad you like the icon :) I've fixed the compiler warnings. If there's interest, I could give the main icon a refresh as well. I'd love to do that.

**Comment by nangtrongvuon** [comment]
There's definitely interest from me for that new icon :)

**Comment by jansol** [comment]
See also xi-editor/xi-editor#51

That issue has a lot of suggestions but sadly no consensus has been reached. I think people expected @raphlinus to make an authoritative decision there but things are quite different now.

**Pull request merged**